### PR TITLE
Remove direct use of object manager

### PR DESCRIPTION
**Pull request by AnshuMishra17**
### Description (*)
Remove the direct use of object manager and replace it with the dependency injection.

### Manual testing scenarios (*)
1. Tried accessing the cms grid listing page and got success.
2. Ran unit tests for cms module.

### CODE REVIEW

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Index.php [hunk 1]
```diff
@@ -26,16 +28,24 @@ class Index extends \Magento\Backend\App\Action implements HttpGetActionInterfac
...
     /**
      * @param Context $context
      * @param PageFactory $resultPageFactory
+     * @param DataPersistorInterface $dataPersistor
      */
     public function __construct(
         Context $context,
-        PageFactory $resultPageFactory
+        PageFactory $resultPageFactory,
+        DataPersistorInterface $dataPersistor
```
Comment (atishgoswami) [inline app/code/Magento/Cms/Controller/Adminhtml/Page/Index.php]:
```suggestion
        DataPersistorInterface $dataPersistor = null
```
To make it an optional backward compatible argument

Comment (AnshuMishra17) [reply app/code/Magento/Cms/Controller/Adminhtml/Page/Index.php]: Thanks @atishgoswami 
I have updated the code as per the suggestion.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5168](https://jira.corp.magento.com/browse/ENGCOM-5168) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Review by orlangur — CHANGES_REQUESTED** [review]
Please squash changes into a single commit so that we have perfectly clean history 😉

**Comment by AnshuMishra17** [comment]
@orlangur I have squashed to single commit.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5168](https://jira.corp.magento.com/browse/ENGCOM-5168) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**